### PR TITLE
RATIS-1554. Fix SnapshotManager RV_RETURN_VALUE_IGNORED_BAD_PRACTICE.

### PR DESCRIPTION
**Pull request by szetszwo**
See https://issues.apache.org/jira/browse/RATIS-1554

### CODE REVIEW

File: ratis-server/src/main/java/org/apache/ratis/server/storage/SnapshotManager.java [hunk 1]
```diff
@@ -121,21 +122,63 @@ public void installSnapshot(StateMachine stateMachine,
...
+
+    // rename stateMachineDir to tmp, if it exists.
+    final File existingDir;
+    if (stateMachineDir.exists()) {
+      File moved = null;
+      try {
+        moved = FileUtils.move(stateMachineDir, TMP + StringUtils.currentDateTime());
+      } catch(IOException e) {
+        LOG.warn("Failed to rename state machine directory " + stateMachineDir.getAbsolutePath()
+            + " to a " + TMP + " directory.  Try deleting it directly.", e);
```
Comment (adoroszlai) [inline ratis-server/src/main/java/org/apache/ratis/server/storage/SnapshotManager.java:159]: Do these warnings intentionally avoid use of placeholders?

**Review by adoroszlai — APPROVED** [review]
Thanks @szetszwo for the patch, LGTM.

File: ratis-server/src/main/java/org/apache/ratis/server/storage/SnapshotManager.java (hunk 1 above)
Comment (szetszwo) [reply ratis-server/src/main/java/org/apache/ratis/server/storage/SnapshotManager.java:159]: You mean "{}"?  Unfortunately, if we use "{}", then the stack trace of the exception won't be printed.

It is easy to have bug like below:
```
LOG.warn{"a: {}, b: {}", a, b, e);
```
The exception e won't be printed at all.

File: ratis-common/src/test/java/org/apache/ratis/util/FileUtilsTest.java [hunk 2]
```diff
@@ -17,25 +17,26 @@
...
-    FileUtils.renameFileToCorrupt(srcFile);
-    srcFile.deleteOnExit();
-    corruptFile.deleteOnExit();
-    assertTrue(corruptFile.exists());
+    Assert.assertTrue(srcFile.exists());
+
+    final File renamed = FileUtils.move(srcFile, ".corrupt");
+    Assert.assertNotNull(renamed);
+    Assert.assertTrue(renamed.exists());
+    Assert.assertFalse(srcFile.exists());
```
[the remark this answers is not in the document]
Comment (szetszwo) [reply ratis-common/src/test/java/org/apache/ratis/util/FileUtilsTest.java:51]: That's a good point!  Let's create the file under the test directory.

**Pull request merged**